### DISCUSSION TITLE
Return null from rpc instead of null-filled json

### DISCUSSION THREAD

**Discussion by martinsellergren** [PostgREST/postgrest#3695] [Q&A] [answered]
I have the function 
```sql
create function "getStory" ("id" uuid) returns "apiStory" language sql 
begin atomic
select
    s.*
from
    "apiStory" s
where
    s."storyId" = "getStory"."id";
end;
```

Calling it with rpc with a non-existent id parameter returns 
```json
{
   "storyId":null,
   "title":null,
   "datePublished":null,
   "dateCreated":null,
   "dateModified":null,
   "authorUserId":null,
   "authorName":null
}
```

but I want it to return null. Is it possible currently? If not, I'd like to request this option.

Motivation: I want my api to return either a valid object, or null.
Note: I'm using supabase.

**Comment by wolfgangwalther** [comment]
You can have your function return `json` directly:

```sql
create function "getStory" ("id" uuid) returns json language sql 
return to_json((
  select
      s
  from
      "apiStory" s
  where
      s."storyId" = "getStory"."id"
));
```

I think this should do what you want it to.

Edit: Slightly adjusted the query.

**Comment by martinsellergren** [comment, marked as answer]
Stumbled upon the solution: Change return type to `record`, instead of the composite type.
